JS/signup.js

```diff
@@ -0,0 +1,130 @@
+// 이메일 유효성 검사
+function validateEmail() {
+  const emailElement = document.getElementById("email");
+  const emailError = document.getElementById("emailError");
+
+// 이메일 정규 표현식
+const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+  // 이메일 형식 검사
+  if (!emailElement.value.trim()) {
+      emailError.innerText = "이메일을 입력해주세요.";
+      emailElement.classList.add("input-error");
+  } else if (!emailRegex.test(emailElement.value)) {
+      emailError.innerText = "잘못된 이메일 형식입니다.";
+      emailElement.classList.add("input-error");
+  } else {
+      emailError.innerText = "";
+      emailElement.classList.remove("input-error");
+  }
+
+  updateSubmitButtonState();
+}
+
+// 닉네임 유효성 검사
+function validateNickname() {
+  const nickNameElement = document.getElementById("nickname");
+  const nicknameError = document.getElementById("nicknameError");
+
+  if (!nickNameElement.value.trim()) {
+    nicknameError.innerText = "닉네임을 입력해주세요.";
+    nickNameElement.classList.add("input-error");
+  } else {
+    nicknameError.innerText = "";
+    nickNameElement.classList.remove("input-error");
+  }
+
+  updateSubmitButtonState();
+}
+
+// 비밀번호 유효성 검사
+function validatePassword() {
+  const passwordElement = document.getElementById("password");
+  const passwordError = document.getElementById("passwordError");
+
+  // 비밀번호 길이 검사
+  if (!passwordElement.value.trim()) {
+      passwordError.innerText = "비밀번호를 입력해주세요.";
+      passwordElement.classList.add("input-error");
+  } else if (passwordElement.value.length < 8) {
```

Comment:
비밀번호가 지금은 8글자까지인데 이후에 바뀔 수 있고, 단순한 숫자값이기에 상수를 정의해서 써주시면 더 좋을 거 같아요!

```suggestion
  const MINIMUM_PW_LENGTH = 8
  
  ....
  
  } else if (passwordElement.value.length < MINIMUM_PW_LENGTH) {
```